### ISSUE TITLE
no-extra-parens handles precedence incorrectly in some cases

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7978]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.4.0
* **npm Version:** 4.0.5

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```yml
rules:
  no-extra-parens: error
parserOptions:
  ecmaVersion: 2016
```

**What did you do? Please include the actual source code causing the issue.**

```js
(new Foo)();

-(foo ** bar);

(-foo) ** bar;
```

**What did you expect to happen?**

I expected no errors to be reported, because removing parentheses from each of those cases results in a change in behavior or a syntax error.

**What actually happened? Please include the actual, raw output from ESLint.**

3 errors were reported:

```
<text>
  1:1  error  Gratuitous parentheses around expression  no-extra-parens
  3:2  error  Gratuitous parentheses around expression  no-extra-parens
  5:1  error  Gratuitous parentheses around expression  no-extra-parens
```
